softlayer/resource_softlayer_file_storage.go

```diff
@@ -594,18 +594,20 @@ func WaitForStorageAvailable(d *schema.ResourceData, meta interface{}) (interfac
 
 			// Check volume status.
 			log.Println("Checking volume status.")
-			resultStr := ""
+			var resultObj interface{}
 			err = sess.DoRequest(
 				"SoftLayer_Network_Storage",
 				"getObject",
 				nil,
 				&sl.Options{Id: &id, Mask: "volumeStatus"},
-				&resultStr,
+				&resultObj,
 			)
```

Comment:
@minsikl It will happen in any case then. However, using .GetObject as above results now in the same call you are doing using DoRequest, so it's not necessary to use DoRequest.